python/cuml/test/dask/test_label_encoder.py

```diff
@@ -0,0 +1,171 @@
+         (cudf.Series(['Tokyo', 'Paris', 'Austin']),
+          cudf.Series([0, 2, 0]),
+          cudf.Series(['Austin', 'Tokyo', 'Austin']),
+          cudf.Series([0, 1, 2, 3])),
+         (cudf.Series(['a', 'b', 'c1']),
+          cudf.Series([2, 1]),
+          cudf.Series(['c1', 'b']),
+          cudf.Series([0, 1, 2, 3])),
+         (cudf.Series(['1.09', '0.09', '.09', '09']),
+          cudf.Series([0, 1, 2, 3]),
+          cudf.Series(['.09', '0.09', '09', '1.09']),
+          cudf.Series([0, 1, 2, 3, 4]))])
+def test_inverse_transform(orig_label, ord_label,
+                           expected_reverted, bad_ord_label,
+                           use_fit_transform, client):
+    n_workers = len(client.has_what())
+    orig_label = dask_cudf.from_cudf(orig_label, npartitions=n_workers)
+    ord_label = dask_cudf.from_cudf(ord_label,
+                                    npartitions=n_workers)
+    expected_reverted = dask_cudf.from_cudf(expected_reverted,
+                                            npartitions=n_workers)
+    bad_ord_label = dask_cudf.from_cudf(bad_ord_label, npartitions=n_workers)
+
+    # prepare LabelEncoder
+    le = LabelEncoder()
+    if use_fit_transform:
+        le.fit_transform(orig_label)
+    else:
+        le.fit(orig_label)
+    assert(le._fitted is True)
+
+    # test if inverse_transform is correct
+    reverted = le.inverse_transform(ord_label)
+    reverted = reverted.compute().reset_index(drop=True)
+    expected_reverted = expected_reverted.compute()
+
+    assert(len(reverted) == len(expected_reverted))
+    assert(len(reverted)
+           == len(reverted[reverted == expected_reverted]))
+    # test if correctly raies ValueError
+    with pytest.raises(ValueError, match='y contains previously unseen label'):
+        le.inverse_transform(bad_ord_label).compute()
+
+
+def test_unfitted_inverse_transform(client):
+    """ Try calling `.inverse_transform()` without fitting first
+    """
+    tmp = cudf.Series(np.random.choice(10, (10,)))
+    df = dask_cudf.from_cudf(tmp, npartitions=len(client.has_what()))
+    le = LabelEncoder()
+
+    with pytest.raises(NotFittedError):
+        le.transform(df)
+
+
+@pytest.mark.parametrize("empty, ord_label",
+                         [(cudf.Series([]), cudf.Series([2, 1]))])
+def test_empty_input(empty, ord_label, client):
+    # prepare LabelEncoder
+    n_workers = len(client.has_what())
+    empty = dask_cudf.from_cudf(empty, npartitions=n_workers)
+    ord_label = dask_cudf.from_cudf(ord_label, npartitions=n_workers)
+    le = LabelEncoder()
+    le.fit(empty)
+    assert(le._fitted is True)
+
+    # test if correctly raies ValueError
+    with pytest.raises(ValueError, match='y contains previously unseen label'):
+        le.inverse_transform(ord_label).compute()
+
+    # check fit_transform()
+    le = LabelEncoder()
+    transformed = le.fit_transform(empty).compute()
+    assert(le._fitted is True)
+    assert(len(transformed) == 0)
```

Comment:
It would be nice to make sure we work correctly with subset/masked arrays (thinking of the other bug that came up for splitting with subset arrays). Something like:

```
x = Series(['a', 'b', 'c', 'd', 'e'])
x_subs = x[[0,3,4]]
res = label_encoder.fit_transform(x_subs)
... ensure res is 0,1,2 and not affected by subset values ...
```

Reply:
Added the test case, but it didn't need that change. This is because - we need to anyway assign the values with 
`ddf_filter = ddf_filter.assign(encoded=encoded.values)` 

Directly assigning to columns won't work because of indices are reset per partitions, and they aren't known earlier.